src/mpc/cli/mpc_phase2_verify_contribution.cpp

```diff
@@ -0,0 +1,151 @@
+#include "mpc_common.hpp"
+#include "snarks/groth16/mpc/phase2.hpp"
+
+using namespace libzeth;
+namespace po = boost::program_options;
+
+namespace
+{
+
+// Usage:
+//   $0 phase2-verify-contribution [<options>] <challenge_file> <response_file>
+//
+// Options:
+//   --transcript <file>     Append contribution, if it is valid
+//   --new-challenge <file>  Write new challenge, if contribution is valid
+class mpc_phase2_verify_contribution : public subcommand
+{
+private:
+    std::string challenge_file;
+    std::string response_file;
+    std::string transcript_file;
+    std::string new_challenge_file;
+
+public:
+    mpc_phase2_verify_contribution()
+        : subcommand("mpc_phase2_verify_contribution")
+        , challenge_file()
+        , response_file()
+        , transcript_file()
+        , new_challenge_file()
+    {
+    }
+
+private:
+    void initialize_suboptions(
+        boost::program_options::options_description &options,
+        boost::program_options::options_description &all_options,
+        boost::program_options::positional_options_description &pos) override
+    {
+        options.add_options()(
+            "transcript",
+            po::value<std::string>(),
+            "Append contribution, if it is valid")(
+            "new-challenge",
+            po::value<std::string>(),
+            "Write new challenge, if contribution is valid");
+        all_options.add(options).add_options()(
+            "challenge_file", po::value<std::string>(), "challenge file")(
+            "response_file", po::value<std::string>(), "response file");
+        pos.add("challenge_file", 1).add("response_file", 1);
+    }
+
+    void parse_suboptions(
+        const boost::program_options::variables_map &vm) override
+    {
+        if (0 == vm.count("challenge_file")) {
```

Comment:
`vm.count("challenge_file") == 0` (ie: switch left hand side and right hand side) or `if(!vm.count("challenge_file")){}`